test/compress/conditionals.js

```diff
@@ -344,7 +344,7 @@ cond_8: {
 
         a = condition ? !0 : !1;
         a = !condition ? !null : !2;
-        a = condition() ? !0 : !-3.5;
+        a = condition() ? !0 : !3.5;
```

Comment:
I was just looking for result symmetry for tests with non-default compress flags. 

Before patch:
```
$ echo 'a = condition() ? !0 : !-3.5;' | uglifyjs -c
a=!!condition();

$ echo 'a = condition() ? !0 : !-3.5;' | uglifyjs -c evaluate=false
a=condition()?!0:!-3.5;
```
After patch:
```
$ echo 'a = condition() ? !0 : !-3.5;' | bin/uglifyjs -c
a=!!condition();

$ echo 'a = condition() ? !0 : !-3.5;' | bin/uglifyjs -c evaluate=false
a=!!condition()||!-3.5;
```
The functions `is_true` and `is_false` within `lib/compress.js` will recognize `!0` and `!1` as booleans because they are so commonly used in the wild. But I did not make that accommodation for negative constants which are actually `AST_UnaryPrefix` expressions with operator `"-"` when `evaluate=false` - not an `AST_Constant` with value `-3.5`.

I wouldn't recommend disabling the `evaluate` compress flag in real-life uglifyjs use.